mso/resource_mso_schema_site_service_graph.go

```diff
@@ -0,0 +1,303 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/container"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSOSchemaSiteServiceGraph() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSOSchemaSiteServiceGraphCreate,
+		Read:   resourceMSOSchemaSiteServiceGraphRead,
+		Update: resourceMSOSchemaSiteServiceGraphUpdate,
+		Delete: resourceMSOSchemaSiteServiceGraphDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOSchemaSiteServiceGraphImport,
+		},
+
+		SchemaVersion: version,
+
+		Schema: (map[string]*schema.Schema{
+			"schema_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"template_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"site_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_graph_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"service_node": &schema.Schema{
```

Comment:
yes.

Reply:
If we have the below config:
```
resource "mso_schema_site_service_graph" "example" {
  schema_id          = mso_schema_site.schema_site_1.schema_id
  site_id            = mso_schema_site.schema_site_1.site_id
  template_name      = "template1"
  service_graph_name = "service_graph1"
  service_node {
    device_dn = data.aci_l4_l7_device.l4_l7_device_1.id
  }
  service_node {
    device_dn = data.aci_l4_l7_device.l4_l7_device_2.id
  }
  service_node {
    device_dn = data.aci_l4_l7_device.l4_l7_device_3.id
  }
}
```
and we change it to ->
```
resource "mso_schema_site_service_graph" "example" {
  schema_id          = mso_schema_site.schema_site_1.schema_id
  site_id            = mso_schema_site.schema_site_1.site_id
  template_name      = "template1"
  service_graph_name = "service_graph1"
  service_node {
    device_dn = data.aci_l4_l7_device.l4_l7_device_1.id
  }
  service_node {
    device_dn = data.aci_l4_l7_device.l4_l7_device_3.id
  }
}
```
There might be a potential problem because in a list of maps, the order of the maps in the list matters. Shall we document this to caution the users? or change it to TypeSet?

Reply:
resolved by documenting this.